saltproc/openmc_depcode.py

```diff
@@ -90,16 +99,140 @@ def __init__(self,
              'settings': str((output_path / 'settings.xml').resolve())}
         self.runtime_matfile = str((output_path / 'materials.xml').resolve())
 
+        self._check_for_material_names(self.template_input_file_path['materials'])
+
+        self._OUTPUTFILE_NAMES = ('depletion_results.h5', 'openmc_simulation_n0.h5',
+                                 'openmc_simulation_n1.h5', 'summary.h5',
+                                 'tallies.out')
+        self._INPUTFILE_NAMES = ('materials.xml', 'geometry.xml', 'tallies.xml')
+
+    def _check_for_material_names(self, filename):
+        """Checks that all materials in the material file
+        have a name.
+
+        Parameters
+        ----------
+        filename : str
+           Path to a materials.xml file
+
+        """
+
+        materials = openmc.Materials.from_xml(filename)
+
+        material_id_to_name = {}
+        for material in materials:
+            if material.name == '':
+                raise ValueError(f"Material {material.id} has no name.")
+
     def read_step_metadata(self):
         """Reads OpenMC's depletion step metadata and stores it in the
         :class:`OpenMCDepcode` object's :attr:`step_metadata` attribute.
         """
+        sp0 = openmc.StatePoint(self.output_path / 'openmc_simulation_n0.h5')
+        sp1 = openmc.StatePoint(self.output_path / 'openmc_simulation_n1.h5')
+        res = Results(self.output_path / 'depletion_results.h5')
+
+        depcode_name, depcode_ver = self.codename, ".".join(list(map(str,sp0.version)))
+        execution_time = sp0.runtime['simulation'] + sp1.runtime['simulation'] + res[0].proc_time + res[1].proc_time
+        sp0.close()
+        sp1.close()
+
+        self.step_metadata['depcode_name'] = depcode_name
+        self.step_metadata['depcode_version'] = depcode_ver
+        self.step_metadata['title'] = ''
+        self.step_metadata['depcode_input_filename'] = ''
+        self.step_metadata['depcode_working_dir'] = str(self.output_path)
+        self.step_metadata['xs_data_path'] = self._find_xs_path()
+        self.step_metadata['OMP_threads'] = -1
+        self.step_metadata['MPI_tasks'] = -1
+        self.step_metadata['memory_optimization_mode'] = -1
+        self.step_metadata['depletion_timestep'] = res[1].time[0]
+        self.step_metadata['execution_time'] = execution_time
+        self.step_metadata['memory_usage'] = -1
```

Comment:
Is there documentation for the contents of this dictionary?

Reply:
Yes. See [this page](https://arfc.github.io/saltproc/master/fileformatspec/databasefile.html)